src/ledger/insert_shred.zig

```diff
@@ -1861,77 +1861,72 @@ fn newlinesToSpaces(comptime fmt: []const u8) [fmt.len]u8 {
 
 const test_shreds = @import("test_shreds.zig");
 const comptimePrint = std.fmt.comptimePrint;
+const TestDB = ledger.tests.TestDB("insert_shred");
+const TestState = ledger.tests.TestState("insert_shred");
 
 fn assertOk(result: anytype) void {
     std.debug.assert(if (result) |_| true else |_| false);
 }
 
 const test_dir = comptimePrint(sig.TEST_DATA_DIR ++ "blockstore/insert_shred", .{});
 
-pub const TestState = struct {
+const ShredInserterTestState = struct {
+    state: *TestState,
     db: BlockstoreDB,
     inserter: ShredInserter,
-    registry: sig.prometheus.Registry(.{}),
 
-    // if this leaks, you forgot to call `TestState.deinit`
-    _leak_check: []const u8,
-
-    // var gpa = std.heap.GeneralPurposeAllocator(.{ .stack_trace_frames = 100 }){};
-    // pub const allocator = gpa.allocator();
-
-    pub const allocator = std.testing.allocator;
-
-    pub fn init(comptime test_name: []const u8) !TestState {
+    pub fn init(comptime test_name: []const u8) !ShredInserterTestState {
         return initWithLogger(test_name, (sig.trace.TestLogger{}).logger());
```

Comment:
it makes sense but is strange to use this way to create a logger when everwhere else we do it a different way ... 

the way it works, we could probably just use it as a global constant in the logger file? `sig.trace.test_logger`

Reply:
>  everwhere else we do it a different way

Do you mean like this?

```zig
const logger = Logger.init(allocator, .warn);
defer logger.deinit();
const some_service = SomeService.init(logger);
```

I think there's a better approach, which is the interface pattern that's used throughout std for things like Allocator, Reader, Writer, and Random.

```zig
const stderr_logger = StandardErrLogger.init(allocator, .warn);
defer stderr_logger.deinit();
const some_service = SomeService.init(stderr_logger.logger());
```

This approach is more flexible, is clearer about the underlying state, and streamlines the interface struct definition.


<br>

> global constant

I can make it a constant, but I'd like to do it in a way that makes it clear that the log level is configurable, and that the constant is only the default. I think it might be slightly confusing if it's at the top level scope of log.zig labelled as the canonical "test_logger". What do you think about this?

```zig
initWithLogger(test_name, sig.trace.TestLogger.default.logger());
```

cf6db5baece72c0a885702d11cb7952b3731273d